build.gradle

```diff
@@ -0,0 +1,66 @@
+group 'spiffe'
+version '0.1-SNAPSHOT'
+
+
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+
+    dependencies {
+        classpath "com.google.protobuf:protobuf-gradle-plugin:0.8.5"
+    }
+}
+
+apply plugin: 'java-library'
+apply plugin: "com.google.protobuf"
+
+ext {
+    grpcVersion = "1.12.0"
+    nettyVersion = "4.1.25.Final"
+}
+
+sourceSets {
+    main {
+        java {
+            srcDirs "build/generated/source/proto/main/grpc"
+            srcDirs "build/generated/source/proto/main/java"
+        }
+    }
+}
+
+protobuf {
+    protoc {
+        artifact = "com.google.protobuf:protoc:3.5.1"
+    }
+    plugins {
+        grpc {
+            artifact = "io.grpc:protoc-gen-grpc-java:${grpcVersion}"
+        }
+    }
+    generateProtoTasks {
+        all()*.plugins {
+            grpc {}
+        }
+    }
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
```

Comment:
You need to explicitely add a dependency on javax.annotation-api for the lib to compile with Java 9 and 10:
`compileOnly group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2'`

This is because GRPC is using the `javax.annotation.Generated` annotation, which is actually part of `java.xml.ws`. It has been moved to a dedicated module from Java 9 on.

For more information have a look at this lengthy discussion about this issue: https://github.com/grpc/grpc-java/issues/3633